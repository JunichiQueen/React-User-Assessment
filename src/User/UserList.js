import React, { Component } from 'react';
import { Table } from 'reactstrap';

import User from './User.js';

export default class UserList extends Component{
    render(){
        return(
            <div>
                <h2 className="text-center">User List</h2>
                <Table className="text-center" striped>
                    <tbody>
                        {this.props.data.map((item) => (
                            <tr key={item._id}><td><User username={item.username}/></td></tr>
                        )
                        )}
                    </tbody>
                </Table>
            </div>
        )
    }
}
