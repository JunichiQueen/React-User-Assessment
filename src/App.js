import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import axios from 'axios';

import logo from './logo.svg';
import './App.css';
import Navbar from './User/Navbar.js';
import Register from './User/Register.js';
import UserList from './User/UserList.js';
import Login from './User/Login.js';

class App extends Component {
  constructor(){
    super();
    this.state={
      data: []
    }
  }
  makeRequest = () => {
    axios
    .get("http://localhost:5000/user/all")
    .then(response => {
      this.setState({
        data: response.data
      })
    })
  }

  componentDidMount = () => {
    this.makeRequest();
  }

  render(){
    return (
      <div>
        <Router>
          <Navbar />
          <Route exact path="/" render={() => <Register getAll={this.makeRequest}/>}/>
          <Route path="/userlist" render={() => <UserList data={this.state.data}/>}/>
          <Route path="/login" render={() => <Login />}/>

        </Router>

        
        


      </div>
    );
  }
}

export default App;
