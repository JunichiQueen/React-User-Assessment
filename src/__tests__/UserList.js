import React from 'react';
import ReactDOM from 'react-dom';
import UserList from '../User/UserList';
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import "../setupTests";
import '../App';

it('renders without crashing', () => {
    shallow(<UserList data={[]}/>);
});
  
it('snapshot app', () => {
    const wrapper = renderer.create(<UserList data={[]}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
});