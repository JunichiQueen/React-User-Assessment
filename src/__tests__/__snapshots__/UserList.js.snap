// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot app 1`] = `
<div>
  <h2
    className="text-center"
  >
    User List
  </h2>
  <table
    className="text-center table table-striped"
  >
    <tbody />
  </table>
</div>
`;
